### PR TITLE
Custom Branding: Relative URL should be allowed

### PR DESCRIPTION
**Pull request by Alankarsharma**
### Description

We have open search running behind a reverse proxy along with other service (Logo is deployed here). From OpenSearch server external URLs are not accessible so if we give this value then Dashboard on startup are not able to fetch the logo and revert to original, if we enter internal URL then server starts properly but logo are not loaded as internal URL is not accessible from external network.

Here if relative URL is passed returning true so that server can start properly and logos are also loaded properly in browser.

Workaround like hostname mapping work only if both internal and external service are deployed on HTTP or HTTPS, but in case of SSL offloading this also fails.
 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
Hello @Alankarsharma, thanks for opening!

Could you please sign the commit? https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/CONTRIBUTING.md#developer-certificate-of-origin.

I merged the branch with `main` to recheck the tests so you might have to pull. But you can amend the commit.

File: src/core/server/rendering/rendering_service.tsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ export class RenderingService {
     };
   }
 
-  public async stop() {}
+  public async stop() { }
```
Comment (abbyhu2000) [inline src/core/server/rendering/rendering_service.tsx:174]: Could you delete the empty space here? it causes a linter error @Alankarsharma

**Comment by abbyhu2000** [comment]
Just a linter error, otherwise LGTM!

File: src/core/server/rendering/rendering_service.tsx (hunk 1 above)
Comment (Alankarsharma) [reply src/core/server/rendering/rendering_service.tsx]: done, thanks for the review

**Comment by abbyhu2000** [comment]
@Alankarsharma Could you please also add the change in the Changelog.md?

**Comment by Alankarsharma** [comment]
> @Alankarsharma Could you please also add the change in the Changelog.md?

Done

**Review by SuZhou-Joe — APPROVED** [review]
Overall, LGTM and relative url should not introduce any security concern like ssrf as we've already do a suffix check beforehand and this config can only be set by cluster admin.

**Review by ananzh — APPROVED** [review]

**Pull request merged**